core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Profiler.scala

```diff
@@ -147,39 +168,59 @@ class Profiler(hadoopConf: Configuration, appArgs: ProfileArgs, enablePB: Boolea
     }
   }
 
-  private def errorHandler(error: Throwable, path: EventLogInfo) = {
-    error match {
+  /**
+   * Process a created ApplicationInfo object and handles all exceptions.
+   */
+  private def profileApp(
+      path: EventLogInfo,
+      index: Int,
+      processSuccessApp: ApplicationInfo => Unit): Unit = {
+    val pathStr = path.eventLog.toString
+    try {
+      val startTime = System.currentTimeMillis()
+      val appOpt = createApp(path, index, hadoopConf)
+      val profAppResult = appOpt match {
+        case Left(FailureApp("skipped", errorMessage)) =>
+          // Case to be skipped
+          progressBar.foreach(_.reportSkippedProcess())
+          SkippedAppResult(pathStr, errorMessage)
+        case Left(FailureApp(_, errorMessage)) =>
+          // Case when other error occurred during ApplicationInfo creation
+          progressBar.foreach(_.reportUnkownStatusProcess())
+          UnknownAppResult(pathStr, "", errorMessage)
+        case Right(app: ApplicationInfo) =>
+          // Case with successful creation of ApplicationInfo
+          processSuccessApp(app)
+          progressBar.foreach(_.reportSuccessfulProcess())
+          val endTime = System.currentTimeMillis()
+          SuccessAppResult(pathStr, app.appId, s"Took ${endTime - startTime}ms to process")
+      }
+      // Log the information to the console
+      profAppResult.logMessage()
+      // Update the appStatusReporter with the result of Application processing
+      appStatusReporter.put(pathStr, profAppResult)
+    } catch {
       case oom: OutOfMemoryError =>
-        logError(s"OOM error while processing large file: ${path.eventLog.toString}." +
-            s" Increase heap size. Exiting ...", oom)
+        logError(s"OOM error while processing large file: $pathStr." +
+            s"Increase heap size.", oom)
         sys.exit(1)
-      case NonFatal(e) =>
-        logWarning(s"Exception occurred processing file: ${path.eventLog.getName}", e)
-      case o: Throwable =>
-        logError(s"Error occurred while processing file: ${path.eventLog.toString}. Exiting ...", o)
+      case o: Error =>
+        logError(s"Error occurred while processing file: $pathStr", o)
         sys.exit(1)
+      case e: Exception =>
+        progressBar.foreach(_.reportFailedProcess())
+        val failureAppResult = FailureAppResult(pathStr,
+          s"Unexpected exception processing log, skipping!")
+        failureAppResult.logMessage(Some(e))
+        appStatusReporter.put(pathStr, failureAppResult)
```

Comment:
can we fix some things with this block?
I prefer that it checks for `NoFatal`  to create a failureApp new object. Anything else should cause an exit... I am not sure what are the cases that can produce `Error`.
Also, this code is pretty much the same we have in Qualification.

Reply:
Also, in future we need to be careful about the exit part when we merge profiling tool to qual tool. It should not exit the qual tool process.

Reply:
thanks! I updated this block. It is similar to Qualification. We can try to refactor something common out. I also want to get this correct as a starting point.